examples/step-45/doc/intro.dox

```diff
@@ -1,72 +1,142 @@
-have the same boundary values:
+suffices to do the actual computation on only one cell of the lattice. To
+be able to proceed this way one has to assume that the computation can be
+periodically extended to the other cells; this requires the solution to
+have a periodic structure.
 
-<img src="http://www.dealii.org/images/steps/developer/step-45.periodic_cells.png" alt="">
+<a name="Procedure"></a>
+<h1>Procedure</h1>
 
-To keep things simple, in this tutorial we will consider an academic,
-simplified problem that allows us to focus on only that part that we are
-interested in here, namely how to set up periodic boundary
-conditions. Specifically, we solve the Poisson problem on a domain, where the
-left and right parts of the boundary are identified. Let $\Omega=(0,1)^2$ and
-consider the problem
-@f{align*}
-   -\Delta u &=
-   \cos(2\pi x)e^{-2x}\cos(2\pi y)e^{-2y}  \qquad &&\text{in }\Omega
-  \\ 
-   u(x,0) &= 0 \qquad &&\text{for }x\in(0,1)\qquad &&\text{(bottom boundary)}
-  \\
-   u(x,1) &= 0 \qquad &&\text{for }x\in(0,1)\qquad &&\text{(top boundary)}
-  \\
-   u(0,y) &= u(1,y) \qquad &&\text{for }y\in(0,1) 
-   \qquad && \text{(left and right boundaries)}
-@f}
-Note that the source term is not symmetric and so the solution would not be
-periodic unless this is explicitly enforced.
+deal.II provides a number of high level functionality to impose periodic
+boundary conditions.
+The general approach to apply periodic boundary conditions consists of
+three steps (see also the
+@ref GlossPeriodicConstraints "Glossary entry on periodic boundary conditions"):
+-# Create a mesh
+-# Gather the periodic faces using GridTools::collect_periodic_faces()
+-# Add the periodicity information to the mesh
+  using parallel::distributed::Triangulation::add_periodicity()
+-# Add periodicity constraints using DoFTools::make_periodicity_constraints()
+The second and third step can be omitted if the Triangulation is not a
+parallel::distributed::Triangulation.
+
+The first step consists of collecting matching periodic faces and store them in
+a <code>std::vector</code> of PeriodicFacePair. This is done with the function
+GridTools::collect_periodic_faces() that can be invoked for example like this:
+@code
+using GridTools::collect_periodic_faces;
+collect_periodic_faces(dof_handler,
+                       b_id1,
+                       b_id2,
+                       direction,
+                       matched_pairs,
+                       offset = <default value>,
+                       matrix = <default value>,
+                       first_vector_components = <default value>);
+@endcode
+
+This call loops over all faces of the container dof_handler on the opposing
+boundaries with boundary indicator b_id1 and b_id2, respecitvely. It matches
+pairs of faces (and dofs) such that the difference between $face_2$ and
+$matrix\cdot face_1+offset$ vanishes in every component apart from direction
```

Comment:
I don't understand how this works. `face_1/2` are objects. What do you mean when you apply a matrix or add an offset to it?